### PR TITLE
repos store: include name of repository in error message

### LINKED ISSUE #45776: RFE: Improve error message 'Syncer: failed to create external service repo: CreateExternalServiceRepo: repo missing source info for external service error message'
**Requested by:** https://github.com/sourcegraph/accounts/issues/6716
**Related:** https://github.com/sourcegraph/customer/issues/1688

#### Feature request description

The error message `Syncer: failed to create external service repo: CreateExternalServiceRepo: repo missing source info for external service error message` shows up on a code host configuration.    Update this message to:

1. Include the repository that triggered this error (or list of repositories)
2. Reduce to a warning if it's not impacting the entire code host

Additionally, include more details in the log message itself so it's easier to debug this.

#### Is your feature request related to a problem? If so, please describe.

This error makes it look like nothing is syncing on this code host.  And, when trying to troubleshoot this error, the logs provide no additional information about which repositories are actually failing (or if the entire code host is not syncing)

/cc @jplahn @ryphil

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes #45776 by including the name of the repository in the error message. That would've saved us hours of debugging.

It's also fine to include this here, since these errors are only ever displayed to admins.

## Test plan

- N/A

### CODE REVIEW

File: internal/repos/store.go [hunk 1]
```diff
@@ -362,7 +362,7 @@ func (s *store) CreateExternalServiceRepo(ctx context.Context, svc *types.Extern
 
 	src := r.Sources[svc.URN()]
 	if src == nil {
-		return errors.New("CreateExternalServiceRepo: repo missing source info for external service")
+		return errors.Newf("CreateExternalServiceRepo: repo %q missing source info for external service", r.Name)
```
Comment (sashaostrikov) [inline internal/repos/store.go:365]: I can see that we have the same situation on the L530 and maybe we can include L446 here as well...

**Review by sashaostrikov — APPROVED** [review]
LGTM, but I think we can fix a couple more error messages in this file in scope of this PR.

**Review by eseliger — COMMENTED** [review]
do we have any take on logging repo names to logging aggregators and such? I don't really know, just throwing it out there

**Comment by mrnugget** [comment]
> do we have any take on logging repo names to logging aggregators and such? I don't really know, just throwing it out there

My take is: if it is a problem, then this error isn't the real problem. Repo names are in every URL and request log, basically. They are on filesystems. Included in a lot of other error messages.

**Pull request merged**